pkgs/build-support/binary-cache.nix

```diff
@@ -0,0 +1,77 @@
+{ stdenv, buildPackages }:
+
+# This function is for creating a flat-file binary cache, i.e. the kind created by
+# nix copy --to file:///some/path and usable as a substituter (with the file:// prefix).
+
+# For example, in the Nixpkgs repo:
+# nix-build -E 'with import ./. {}; mkBinaryCache { rootPaths = [hello]; }'
+
+{ name ? "binary-cache"
+, rootPaths
+}:
+
+stdenv.mkDerivation {
+  inherit name;
+
+  __structuredAttrs = true;
+
+  exportReferencesGraph.closure = rootPaths;
```

Comment:
Isn't `exportReferencesGraph` [supposed to be a list](https://nixos.org/manual/nix/unstable/language/advanced-attributes.html#adv-attr-exportReferencesGraph) rather than an attrset `{closure = ...}`?